files/02_amend_replication_agreements.sh

```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/env bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+TOPOLOGY_SUFFIXES="domain ca"
+
+# Enable replication of the krblastsuccessfulauth timestamps.  By
+# default these timestamps *are not* replicated between servers to
+# avoid a replication storm, but given our limited number of users we
+# should be OK.  See here for more details:
+# https://pagure.io/freeipa/issue/9821
+function enable_last_successful_auth_replication {
+  for suffix in $TOPOLOGY_SUFFIXES; do
+    # Get all the topology segments for the given topology suffix.  We
+    # will need to update each one.
+    topology_segments=$(ipa topologysegment-find "$suffix" --pkey-only \
+      | sed --quiet "s/^[[:blank:]]*Segment name:[[:blank:]]*\(.*\)$/\1/p")
+    for segment in $topology_segments; do
+      # Get detailed information about the topology segment so we can
+      # modify it.
+      cmd_output=$(ipa topologysegment-find "$suffix" --all \
+        --name="$segment" --raw)
+      old_repl_attr=$(sed --quiet \
+        "s/^[[:blank:]]*nsDS5ReplicatedAttributeList:[[:blank:]]*\(.*\)$/\1/p" \
+        <<< "$cmd_output")
+      new_repl_attr=${old_repl_attr// krblastsuccessfulauth/}
+      old_repl_attr_total=$(sed --quiet \
+        "s/^[[:blank:]]*nsDS5ReplicatedAttributeListTotal:[[:blank:]]*\(.*\)$/\1/p" \
+        <<< "$cmd_output")
+      new_repl_attr_total=${old_repl_attr_total// krblastsuccessfulauth/}
```

Comment:
It might be nice to add a brief comment to make it clear that this is where we remove `krblastsuccessfulauth` from the list of replication attributes.

Reply:
Please see commit 97f223a.